core-primitives/stf-executor/src/executor.rs

```diff
@@ -414,14 +415,18 @@ pub fn shards_key_hash() -> Vec<u8> {
 /// assumes a common structure of sgx_runtime and extracts interesting metrics
 /// while this may not be the best abstraction, it avoids circular dependencies
 /// with app-libs and will be suitable in 99% of cases
-fn gather_runtime_metrics<State>(state: &State) -> RuntimeMetrics
+fn gather_runtime_metrics<State>(state: &State, decimals: u8) -> RuntimeMetrics
 where
 	State: SgxExternalitiesTrait + Encode,
 {
 	// prometheus has no support for NaN, therefore we fall back to -1
 	let total_issuance: f64 = state
 		.get(&storage_value_key("Balances", "TotalIssuance"))
-		.map(|v| Balance::decode(&mut v.as_slice()).map(|b| b as f64).unwrap_or(-1.0))
+		.map(|v| {
+			Balance::decode(&mut v.as_slice())
+				.map(|b| (b as f64) / 10f64.powi(decimals as i32))
```

Comment:
division is safe because 10^X can never be zero

Reply:
but balances as f64 is unsafe ;)

Reply:
it is safe, it's just not precise. for metrics that's fine IMO